server/controllers/tasks/lib/build_task_docs.coffee

```diff
@@ -2,49 +2,82 @@ __ = require('config').universalPath
+    { labels, authorId } = authorWorksData
+    # Todo : check every labels to turn entity into redirection
+    unless canBeRedirected suggestions, labels[0] then return occurences
+    # assume first occurence is the right one to merge into
+    # only one suggestion necessary to merge
+    # occurences of first suggestion picked
+    turnIntoRedirection reconcilerUserId, authorId, occurences[0].uri
+    []
+
+filterSuggestions = (occurences, suggestions)->
+  # create a task for every suggestions
+  unless _.some(_.flattenDeep(occurences)) then return suggestions
+  # create tasks only for suggestions with occurences
+  suggestions.filter suggestionsWithOccurences occurences
+
+createTasksDocs = (authorWorksData, occurences) ->
+  return (suggestions) ->
+    relationScore = calculateRelationScore suggestions
+    return suggestions.map create authorWorksData, relationScore, occurences
+
+canBeRedirected = (suggestions, workLabel) ->
+  # several suggestions == has homonym
+  unless suggestions.length == 1 then return false
+  workLabel.length > 12
+
+suggestionsWithOccurences = (occurences)->
+  return (suggestions)->
+    suggestionsUris = _.pluck suggestions, 'uri'
+    occurencesUris = _.pluck occurences, 'uri'
+    _.intersection suggestionsUris, occurencesUris
+
+getOccurences = (authorWorksData)->
+  return (suggestion)->
+    { labels, langs } = authorWorksData
+    hasWorksLabelsOccurrence suggestion.uri, labels, langs
+    .then (occurences)->
+      if _.isEmpty(_.flattenDeep(occurences)) then return []
+      return
+        uri: suggestion.uri
+        occurences: occurences
+
+create = (authorWorksData, relationScore, suggestionsOccurences)->
+  return (suggestion)->
+    { authorId } = authorWorksData
+    suggestionOccurences = _.filter(suggestionsOccurences, {'uri': suggestion.uri})
+    unless _.isEmpty suggestionOccurences
+      occurences = suggestionOccurences[0]['occurences']
+
+    return
       type: 'deduplicate'
-      suspectUri: prefixifyInv authorWorksData.authorId
-      suggestionUri: suggestionEntity.uri
-      lexicalScore: suggestionEntity._score
+      suspectUri: prefixifyInv(authorId)
+      suggestionUri: suggestion.uri
+      lexicalScore: suggestion._score
       relationScore: relationScore
-      hasEncyclopediaOccurence: hasOccurence
-
-getAuthorWorksData = (authorId)->
-  entities_.byClaim 'wdt:P50', "inv:#{authorId}", true, true
-  .then (works)->
-    # works = [
-    #   { labels: { fr: 'Matiere et Memoire'} },
-    #   { labels: { en: 'foo' } }
-    # ]
-    base = { authorId, labels: [], langs: [] }
-    worksData = works.reduce aggregateWorksData, base
-    worksData.langs = _.uniq worksData.langs
-    return worksData
-
-aggregateWorksData = (worksData, work)->
-  for lang, label of work.labels
-    worksData.labels.push label
-    worksData.langs.push lang
-  return worksData
+      hasEncyclopediaOccurence: occurences or {}
```

Comment:
`hasEncyclopediaOccurence` was hinting to a boolean value, this could be renamed something like `encyclopediaOccurences`, or, for non-encyclopedic future-proofness, something like `externalSourcesOccurrences`